bundles/org.openhab.binding.yamahamusiccast/src/main/java/org/openhab/binding/yamahamusiccast/internal/YamahaMusiccastHandler.java

```diff
@@ -822,6 +853,109 @@ private void updatePresets(int value) {
+                                    new StateOption(String.valueOf(presetCounter), "#" + presetCounter + " " + text));
+                        }
+                    }
+                }
+                for (Channel channel : getThing().getChannels()) {
+                    ChannelUID channelUID = channel.getUID();
+                    channelWithoutGroup = channelUID.getIdWithoutGroup();
+                    if (isLinked(channelUID)) {
+                        switch (channelWithoutGroup) {
+                            case CHANNEL_SELECTPRESET_DAB:
+                                stateDescriptionProvider.setStateOptions(channelUID, optionsPresets);
+                                break;
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private void updatePresetsFM() {
+        String inputText = "";
+        int presetCounter = 0;
+        tmpString = getPresetInfoTuner(this.host, "fm");
+
+        PresetInfo presetinfo = gson.fromJson(tmpString, PresetInfo.class);
+        if (presetinfo != null) {
+            String responseCode = presetinfo.getResponseCode();
+            if ("0".equals(responseCode)) {
+                List<StateOption> optionsPresets = new ArrayList<>();
+                inputText = getLastInput();
+                if (inputText != null) {
+                    for (JsonElement pr : presetinfo.getPresetInfo()) {
+                        JsonObject presetObject = pr.getAsJsonObject();
+                        presetCounter = presetCounter + 1;
+                        String text = presetObject.get("text").getAsString();
+                        if (!"".equals(text)) {
+                            optionsPresets.add(
+                                    new StateOption(String.valueOf(presetCounter), "#" + presetCounter + " " + text));
+                        }
+                    }
+                }
+                for (Channel channel : getThing().getChannels()) {
+                    ChannelUID channelUID = channel.getUID();
+                    channelWithoutGroup = channelUID.getIdWithoutGroup();
+                    if (isLinked(channelUID)) {
+                        switch (channelWithoutGroup) {
+                            case CHANNEL_SELECTPRESET_FM:
+                                stateDescriptionProvider.setStateOptions(channelUID, optionsPresets);
+                                break;
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private void updateTunerPlayer() {
+        tmpString = getPlayInfoTuner(this.host);
+
+        @Nullable
+        PlayInfoTuner targetObject = gson.fromJson(tmpString, PlayInfoTuner.class);
+        if (targetObject != null) {
+            String responseCode = targetObject.getResponseCode();
+
+            if ("0".equals(responseCode)) {
+                int dabPreset = targetObject.getDAB().getPreset();
```

Comment:
Such calls a really dangerous, since getDAB() may return null as the targetObject is a DTO and the field might be missing in the response. 
This is for example the case for devices with FM and without DAB.
Please never ever assume something is not null on a DTO if you are not a hundred percent sure!

I have experienced serious issues as described in my fix https://github.com/openhab/openhab-addons/pull/19895.